src/ml/neural_net/image_augmentation.cpp

```diff
@@ -117,5 +117,98 @@ image_augmenter::result resize_only_image_augmenter::prepare_images(
+    image_util::copy_image_to_memory(source_batch[i].image, img.data(),
+                                     {input_width * c, c, 1},
+                                     {input_height, input_width, c}, true);
+
+    std::transform(img.begin(), img.end(), img.begin(),
+                   [](float pixel) -> float { return pixel / 255; });
+    shared_float_array image_to_aug =
+        shared_float_array::wrap(img, {input_height, input_width, c});
+    images_to_aug.push_back(image_to_aug);
+    std::vector<float> annotation(source_batch[i].annotations.size() * 6);
+    size_t x = 0;
+    for (size_t j = 0; j < source_batch[i].annotations.size(); j++) {
+      annotation[x] = source_batch[i].annotations[j].identifier;
+      annotation[x + 1] = source_batch[i].annotations[j].bounding_box.x;
+      annotation[x + 2] = source_batch[i].annotations[j].bounding_box.y;
+      annotation[x + 3] = source_batch[i].annotations[j].bounding_box.height;
+      annotation[x + 4] = source_batch[i].annotations[j].bounding_box.width;
+      annotation[x + 5] = source_batch[i].annotations[j].confidence;
+      x = (j + 1) * 6;
+    }
+    shared_float_array ann_to_aug = shared_float_array::wrap(
+        annotation, {source_batch[i].annotations.size() * 6});
+    ann_batch.push_back(ann_to_aug);
+    std::vector<float> predictions(source_batch[i].predictions.size() * 6);
+    size_t y = 0;
+    for (size_t k = 0; k < source_batch[i].predictions.size(); k++) {
+      predictions[y] = source_batch[i].predictions[k].identifier;
+      predictions[y + 1] = source_batch[i].predictions[k].bounding_box.x;
+      predictions[y + 2] = source_batch[i].predictions[k].bounding_box.y;
+      predictions[y + 3] = source_batch[i].predictions[k].bounding_box.height;
+      predictions[y + 4] = source_batch[i].predictions[k].bounding_box.width;
+      predictions[y + 5] = source_batch[i].predictions[k].confidence;
+      y = (k + 1) * 6;
+    }
+
+    turi::neural_net::shared_float_array pred_to_aug = shared_float_array::wrap(
+        predictions, {source_batch[i].predictions.size() * 6});
+
+    pred_batch.push_back(pred_to_aug);
+  }
+  intermediate_labeled_image input_to_tf_aug;
+  input_to_tf_aug.images = images_to_aug;
+  input_to_tf_aug.annotations = ann_batch;
+  input_to_tf_aug.predictions = pred_batch;
+  intermediate_result augmented_data =
+      prepare_augmented_images(input_to_tf_aug);
+  res.image_batch = augmented_data.images;
+  std::vector<std::vector<image_annotation>> annotations_per_batch;
+  for (size_t a = 0; a < n; a++) {
+    std::vector<image_annotation> annotations_per_image;
+    const size_t* sh = augmented_data.annotations[a].shape();
+
+    for (size_t b = 0; b < sh[0]; b++) {
+      image_annotation annotation;
+      image_box bbox;
+      std::cout << a << b;
+      const float* ptr = augmented_data.annotations[a][b].data();
+      annotation.identifier = static_cast<int>(ptr[0]);
+      bbox.x = static_cast<float>(ptr[1]);
```

Comment:
Why are these `static_cast`s needed?

Reply:
They all are `shared_float_arrays` need to be casted in as float/int.